perf/roles/trex/tasks/main.yml

```diff
@@ -1,37 +1,58 @@
 ---
 
-- name: install build dependencies
+- name: install runtime dependencies
   dnf:
     name:
-      - gcc
-      - gcc-c++
-      - git
       - python3
-      - rdma-core-devel
-      - zlib-devel
+      - ca-certificates
+      - python3-cryptography
     state: installed
   tags: install
 
+- name: Update CA certificates
+  command: update-ca-trust enable
+
+- name: Generate private key
+  community.crypto.openssl_privatekey:
+    path: /etc/pki/tls/private/private.key
+    size: 2048
+  register: private_key_result
+
+- name: Generate an OpenSSL Certificate Signing Request
+  community.crypto.openssl_csr:
+    path: /etc/pki/tls/private/cert.csr
+    privatekey_path: /etc/pki/tls/private/private.key
+    common_name: www.ansible.com
+
+- name: Generate a Self Signed OpenSSL certificate
+  community.crypto.x509_certificate:
+    path: /etc/pki/ca-trust/source/anchors/certificate.crt
+    privatekey_path: /etc/pki/tls/private/private.key
+    csr_path: /etc/pki/tls/private/cert.csr
+    provider: selfsigned
+
+- name: Extract and update trusted CA's
+  command: update-ca-trust extract
```

Comment:
I don't understand what this is for? Why are you generating a self-signed certificate?

Reply:
Updated ... Earlier I got confused because I was not aware of how can I export the certificate ... Now updated ...